packages/zwave-js/src/lib/commandclass/SceneActuatorConfigurationCC.ts

```diff
@@ -0,0 +1,445 @@
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+// All the supported commands
+export enum SceneActuatorConfigurationCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+}
+
+export function getLevelValueID(
+	endpoint: number | undefined,
+	sceneId: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Scene Actuator Configuration"],
+		endpoint,
+		property: "level",
+		propertyKey: sceneId,
+	};
+}
+
+export function getDimmingDurationValueID(
+	endpoint: number | undefined,
+	sceneId: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Scene Actuator Configuration"],
+		endpoint,
+		property: "dimmingDuration",
+		propertyKey: sceneId,
+	};
+}
+
+function persistSceneActuatorConfig(
+	this: SceneActuatorConfigurationCC,
+	sceneId: number,
+	level: number,
+	dimmingDuration: Duration,
+) {
+	const levelValueId = getLevelValueID(this.endpointIndex, sceneId);
+	const dimmingDurationValueId = getDimmingDurationValueID(
+		this.endpointIndex,
+		sceneId,
+	);
+	const valueDB = this.getValueDB();
+
+	if (!valueDB.hasMetadata(levelValueId)) {
+		valueDB.setMetadata(levelValueId, {
+			...ValueMetadata.UInt8,
+			label: `Level (${sceneId})`,
+		});
+	}
+	if (!valueDB.hasMetadata(dimmingDurationValueId)) {
+		valueDB.setMetadata(dimmingDurationValueId, {
+			...ValueMetadata.Duration,
+			label: `Dimming duration (${sceneId})`,
+		});
+	}
+
+	valueDB.setValue(levelValueId, level);
+	valueDB.setValue(dimmingDurationValueId, dimmingDuration);
+
+	return true;
+}
+
+@API(CommandClasses["Scene Actuator Configuration"])
+export class SceneActuatorConfigurationCCAPI extends CCAPI {
+	public supportsCommand(
+		cmd: SceneActuatorConfigurationCommand,
+	): Maybe<boolean> {
+		switch (cmd) {
+			case SceneActuatorConfigurationCommand.Get:
+				return this.isSinglecast();
+			case SceneActuatorConfigurationCommand.Set:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	): Promise<void> => {
+		if (propertyKey == undefined) {
+			throwMissingPropertyKey(this.ccId, property);
+		} else if (typeof propertyKey !== "number") {
+			throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+		}
+		if (property === "level") {
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+
+			// We need to set the dimming duration along with the level
+			const node = this.endpoint.getNodeUnsafe()!;
+			const dimmingDuration = node.getValue<Duration>(
+				getDimmingDurationValueID(this.endpoint.index, propertyKey),
+			);
+			// If dimmingDuration is missing, we must abort
+			if (dimmingDuration == undefined) {
+				throw new ZWaveError(
+					`The "dimmingDuration" property cannot be changed before the level is known!`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
```

Comment:
We're stuck in an infinite loop here, because you also cannot set the dimmingDuration without the level.

I feel like I'm making you run in circles, but it seems like we do need to treat dimmingDuration as a meta property that needs to be set along with the level (#1321)

Reply:
I'm removing the option to set the `dimmingDuration` alone and reallowing a default of `0 seconds` to be set for the `dimmingDuration` if the user attempts to do a SET_VALUE on `level` before a `dimmingDuration` has been set.  That seems like a reasonable default.